### PR TITLE
Milestone 4: Create join tables

### PR DESCRIPTION
**Pull request by zdnahom**
## Milestone 4: Create join tables
In this milestone I : 
- Created vets table
- Created specializations table which connects vets and species tables
- Created visits table which connects vets and animals tables
- Added queries based on the requirements

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -231,4 +231,114 @@ FROM
...
+SELECT
+    COUNT(vets.name) AS vets_without_specialization
+FROM
+    animals
+    JOIN visits ON visits.animals_id = animals.id
+    JOIN vets ON vets.id = visits.vets_id
+    LEFT JOIN specializations ON specializations.vets_id = vets.id
+    AND specializations.species_id = animals.species_id
+WHERE
+    specializations.species_id IS NULL;
```
Comment (Deepakdanger) [inline queries.sql]: - [ ] Here we have to find 
- * How many times visits where with a vet that did not specialize in that animal's species?

```sql
  SELECT count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE animals.species_id NOT IN (SELECT species_id FROM specializations WHERE vet_id = vets.id)
```

File: queries.sql [hunk 2]
```diff
@@ -231,4 +231,114 @@ FROM
...
+FROM
+    animals
+    JOIN visits ON animals.id = visits.animals_id
+    JOIN vets ON vets.id = visits.vets_id
+GROUP BY
+    animals.id
+ORDER BY
+    total_visits DESC
+limit
+    1;
```
Comment (Deepakdanger) [inline queries.sql]: - [ ] Here we have to find, What animal has the most visits to vets?
```
  SELECT animals.name, count(animals.name)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  ORDER BY count(animals.name) DESC
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @zdnahom , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (zdnahom) [reply queries.sql]: That's a simple way to do it. Thanks for the review @Deepakdanger, I will definitely follow your advice 🙌 .

File: queries.sql (hunk 2 above)
Comment (zdnahom) [reply queries.sql]: I will update this too😉

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @zdnahom, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**